### PR TITLE
Various bug fixes and improvements.

### PR DESCRIPTION
**Pull request by bitonic**
I suggest to review commit-by-commit, they do one thing each and they should explain well what the fix is.

There are no tests yet -- I hope to get around to it soon.

---
This change is [Reviewable](https://reviewable.io/reviews/codedownio/aeson-typescript/17)

### CODE REVIEW

File: src/Data/Aeson/TypeScript/Types.hs [hunk 1]
```diff
@@ -93,12 +93,14 @@ data FormattingOptions = FormattingOptions
   -- ^ Function applied to generated interface names
   , typeNameModifier :: String -> String
   -- ^ Function applied to generated type names
+  , exportTypes :: Bool
```
Comment (thomasjm) [inline src/Data/Aeson/TypeScript/Types.hs:96]: Could you add a documentation string here?

Comment (thomasjm) [reply src/Data/Aeson/TypeScript/Types.hs:96]: Also curious about the need for this -- I always put my types in `.d.ts` files, is there some reason you want to put them in an ES6 module?

Comment (bitonic) [reply src/Data/Aeson/TypeScript/Types.hs:96]: I generate automatically not only the types, but also automatically generated code that builds a data structure that holds `fetch` invocations with the right request and response type, through https://github.com/bitonic/solga/tree/francesco/solga-typescript . I store the generated types alongside this definition, and therefore I need the exports.

**Comment by bitonic** [comment]
@thomasjm thanks, I'll take a look at those soon.

**Comment by thomasjm** [comment]
The latest commits look good @bitonic . I'd love to get this merged and get a new release out with these improvements. Are you planning to get the tests passing? Thanks!

**Comment by thomasjm** [comment]
Working on integrating the remaining changes from this PR:

* 92fdfba ~~2020-07-08 | Add instance for `Word8` (bitonic/francesco) [Francesco Mazzoli]~~
* b61824c 2020-05-22 | add parent types / generic instances for unary records [Francesco Mazzoli]
* f6e51e4 2020-04-15 | cabal file update [Francesco Mazzoli]
* a751b8b 2020-04-15 | Allow to access `TypeScript` types from outside [Francesco Mazzoli]
* 13822f0 2020-04-08 | allow to export types easily [Francesco Mazzoli]
* 2a86676 2020-04-08 | use mapped types to represent maps [Francesco Mazzoli]
* 56f7de2 2020-04-08 | add some missing container instances [Francesco Mazzoli]
* 7645e67 2020-04-08 | when the type refer to a single interface, refer to it correctly [Francesco Mazzoli]
* 6ffc83e 2020-04-08 | honor `unwrapUnaryRecords` for unary records [Francesco Mazzoli]
* cd8e46a 2020-04-08 | include immediate parents for containers... [Francesco Mazzoli]

**Comment by thomasjm** [comment]
Okay, I think that's everything! I'll get a release out with all this stuff soon. Thanks again.

**Pull request closed without merging**